### PR TITLE
Added dev-ddb to build.boot, updated building documentation

### LINKED ISSUE #400: Dev-ddb no longer supported for local development
## BEFORE SUBMITTING AN ISSUE CHECK THE FOLLOWING:
- [x] I checked existing issues.
- [x] This is not a support question (which can be answered on Slack).
- [x] This issue is for the API / Game Engine code base (not [Web Client](https://github.com/willowtreeapps/wombats-web-client), [Lambda](https://github.com/willowtreeapps/wombats-lambda), or [Bots](https://github.com/willowtreeapps/wombats-bots))

## ISSUE DESCRIPTION
The dev-ddb launch option no longer functions correctly. This functionality was temporarily removed to fix build issues on development, and must be added back in. 

See build.boot "is-local?" function.

### PR DESCRIPTION
**Pull request by elibosley**
- Fixes #400, adds documentation about the build process
- Fixes an issue that @Daniel-Hoerauf found with loading private configuration files from two locations
# PR for issue #400.

## Ready for a PR?
 - [x] I have updated the CHANGELOG with an entry including a description, a link to my profile, and a link to the issue ticket. This change is filed under an Enhancement or Bug Fix, is shown under the relevant release tag name, and is included in my PR branch.

### CODE REVIEW

**Comment by elibosley** [comment]
@dehli This should be good to go, and patches a couple of minor issues

File: build.boot [hunk 1]
```diff
@@ -216,10 +215,9 @@ Type \"Yes\" to confirm."
   []
   (let [file-location-dev (str (System/getProperty "user.dir") "/config/config.edn")
         file-location-prod (str (System/getProperty "user.home") "/.wombats/config.edn")]
-    (when (.exists (io/as-file file-location-dev))
-      file-location-dev)
-    (when (.exists (io/as-file file-location-prod))
-      file-location-prod)))
+    (cond
+      (.exists (io/as-file file-location-dev)) file-location-dev
+      (.exists (io/as-file file-location-prod)) file-location-prod)))
```
Comment (dehli) [inline build.boot]: Order should be swapped so that it `file-location-prod` has higher priority.

File: src/wombats/components/configuration.clj [hunk 2]
```diff
@@ -43,12 +44,16 @@
   []
   (let [file-location-dev (str (System/getProperty "user.dir") "/config/config.edn")
         file-location-prod (str (System/getProperty "user.home") "/.wombats/config.edn")]
-    (when (.exists (io/as-file file-location-dev))
-      (println "Using config at /config/config.edn")
-      file-location-dev)
-    (when (.exists (io/as-file file-location-prod))
-      (println "Using config at ~/.wombats/config.edn")
-      file-location-prod)))
+    (cond
```
Comment (dehli) [inline src/wombats/components/configuration.clj]: Order should be swapped here as well so the `prod` location has higher priority.

**Comment by elibosley** [comment]
@dehli This should be good to go!

**Comment by dehli** [comment]
Actually, one last suggestion would be to create a function that pulls out the config file and use that in `build.boot`.

**Comment by elibosley** [comment]
@dehli It's currently duplicated because one of the functions has those extra log items - would you still de-duplicate them?

**Comment by dehli** [comment]
Yep. You can still accomplish logging by doing something like:

```
(let [config-file (get-config-file)]
  (if config-file
    (log/info (str "Using private config-file " config-file)
    (log/info "Not using a private config-file"))
  config-file)
```

**Pull request merged**